src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import ClassNames from 'classnames';
+
+import styles from '@/components/SearchInput/SearchInput.css';
+
+import IconSearch from '@/assets/images/icons/web/search-simple.svg';
+import IconClose from '@/assets/images/icons/web/close.svg';
+
+type Size = 'large' | 'medium';
+type State = '' | 'valid' | 'invalid';
+
+interface Props {
+  description?: string;
+  disabled?: boolean;
+  label?: string;
+  message?: string;
+  onChange: () => void;
+  clearInputClick: () => void;
+  placeholder?: string;
+  state?: State;
+  size: Size;
+  value?: string;
+  variablesClassName?: string;
+}
+
+const SearchInput: React.FC<Props> = props => {
+  const {
+    clearInputClick,
+    description,
+    disabled,
+    label,
+    onChange,
+    message,
+    placeholder,
+    size,
+    state,
+    value,
+    variablesClassName,
+    ...searchInputProps
+  } = props;
+  const sizeClass = `search--${size}`;
+  const statusClass = state ? `search--${state}` : '';
+  const sizeInputClass = `search-input--${size}`;
+  const messageClass = 'search-message';
```

Comment:
does it need to be a variable?

Reply:
I think it better to view and being more organized

Reply:
@Rheniery sorry if I didn't mention my reasoning before, my reasoning is that having all the classes here it makes it hard for us to know which classes are static and which ones go through some processing.

Reply:
Ohh, I thought it would be a pattern to follow, okay... I'll change it :)

Reply:
This way any class that needs some processing is a variable, and the ones that do not need are simply classes in the HTML code